contracts/contracts_test.go

```diff
@@ -50,3 +54,41 @@ func TestUnmarshalCasmClass(t *testing.T) {
 	assert.Equal(t, casmClass.EntryPointByType.External[1].Offset, 130)
 	assert.Equal(t, casmClass.EntryPointByType.External[1].Builtins[0], "range_check")
 }
+
+// TestPrecomputeAddress tests the PrecomputeAddress function.
+//
+// It calls the PrecomputeAddress with predefined parameter values and compares the result with predefined expected results.
+// The function uses the 'require' .NoError and .Equal functions from the github.com/stretchr/testify/assert package to perform the assertions.
+// It is a test function and is meant to be used with the Go testing framework.
+//
+// Parameters:
+// - t: The testing.T instance for running the test
+// Returns:
+//
+//	none
+func TestPrecomputeAddress(t *testing.T) {
+
+	deployerAddress := utils.TestHexToFelt(t, "0xdeadbeef")
+	salts := []*felt.Felt{
+		utils.TestHexToFelt(t, "0xaaaaaaaa"),
+		utils.TestHexToFelt(t, "0xbbbbbbbb"),
+		utils.TestHexToFelt(t, "0xcccccccc"),
+	}
+	classHash := utils.TestHexToFelt(t, "0xdeadbeef")
+	constructorCalldata := []*felt.Felt{
+		deployerAddress,
+	}
+
+	expectedPrecomputedAddress := []*felt.Felt{
+		utils.TestHexToFelt(t, "0x3791840c6746577f27f4a5766a6862b7486f0a254d34e0d2f0a5d238160cf10"),
+		utils.TestHexToFelt(t, "0x479b5192d4af71279a92eddb09e4dcfdc00751ccb27da0e9c4f9b2312318242"),
+		utils.TestHexToFelt(t, "0x7a824044120ac54527d701e18b2eea6e11fb0a8f070a21acd13a7a4062065b4"),
+	}
```

Comment:
Where did you get these? It would also be nice to have a test case that generates an address that exists on-chain

Reply:
I generated them from starknet.js. Nice idea, I'll do it